client/web/src/site-admin/SiteAdminExternalServicesArea.tsx

```diff
@@ -0,0 +1,113 @@
+import { lazyComponent } from '@sourcegraph/shared/src/util/lazyComponent'
+import { LoadingSpinner } from '@sourcegraph/wildcard'
+
+import { codeHostExternalServices, nonCodeHostExternalServices } from '../components/externalServices/externalServices'
+
+import { SITE_EXTERNAL_SERVICE_CONFIG } from './backend'
+
+const ExternalServicesPage = lazyComponent(
+    () => import('../components/externalServices/ExternalServicesPage'),
+    'ExternalServicesPage'
+)
+const ExternalServicePage = lazyComponent(
+    () => import('../components/externalServices/ExternalServicePage'),
+    'ExternalServicePage'
+)
+
+const AddExternalServicesPage = lazyComponent(
+    () => import('../components/externalServices/AddExternalServicesPage'),
+    'AddExternalServicesPage'
+)
+
+interface Props
+    extends RouteComponentProps<{}>,
+        ThemeProps,
+        TelemetryProps,
+        PlatformContextProps,
+        SettingsCascadeProps {
+    authenticatedUser: AuthenticatedUser
+}
+
+export const SiteAdminExternalServicesArea: React.FunctionComponent<React.PropsWithChildren<Props>> = ({
+    match,
+    ...outerProps
+}) => {
+    const { data, error, loading } = useQuery<SiteExternalServiceConfigResult, SiteExternalServiceConfigVariables>(
+        SITE_EXTERNAL_SERVICE_CONFIG,
+        {}
+    )
+
+    if (error && !loading) {
+        return <ErrorAlert error={error} />
+    }
+
+    if (loading && !error) {
+        return <LoadingSpinner />
+    }
+
+    if (!data) {
+        return null
+    }
+
+    return (
+        <Switch>
+            <Route
+                path={match.url}
+                render={props => (
+                    <ExternalServicesPage
+                        {...outerProps}
+                        {...props}
+                        routingPrefix="/site-admin"
+                        afterDeleteRoute="/site-admin/repositories?repositoriesUpdated"
+                        externalServicesFromFile={data?.site?.externalServicesFromFile}
+                        allowEditExternalServicesWithFile={data?.site?.allowEditExternalServicesWithFile}
+                    />
+                )}
+                exact={true}
+            />
+            <Route path={match.url + '/add'} render={() => <Redirect to="new" />} exact={true} />
+            <Route
+                path={match.url + '/new'}
+                render={props => (
+                    <AddExternalServicesPage
+                        {...outerProps}
+                        {...props}
+                        routingPrefix="/site-admin"
+                        afterCreateRoute="/site-admin/repositories?repositoriesUpdated"
+                        codeHostExternalServices={codeHostExternalServices}
+                        nonCodeHostExternalServices={nonCodeHostExternalServices}
+                    />
+                )}
+                exact={true}
+            />
+            <Route
+                path={match.url + '/:id'}
```

Comment:
nit, this would look better IMO
```suggestion
                path={`${match.url}/:id`}
```

Reply:
Really?! We have opposite aesthetic preferences then

Reply:
Okay, turns out there's a technical reason for this?! https://github.com/sourcegraph/sourcegraph/pull/42984/files#r995741960

Reply:
no, the removal of &Props is what fixed it I think.